invoiceregistry_test.go

```diff
@@ -225,27 +308,17 @@ func TestAutoSettle(t *testing.T) {
 
 	// Add invoice.
 	invoice, preimage := c.createInvoice(1, time.Hour)
+	hash := preimage.Hash()
 
 	// Subscribe to updates for invoice.
 	acceptChan, cancelAcceptUpdates := c.subscribeAccept()
 
 	resolved := make(chan struct{})
-	c.registry.NotifyExitHopHtlc(&registryHtlc{
-		rHash:         preimage.Hash(),
-		amtPaid:       lnwire.MilliSatoshi(c.testAmt),
-		expiry:        100,
-		currentHeight: 0,
-		resolve: func(r HtlcResolution) {
-			close(resolved)
-		},
-		payload: &testPayload{
-			amt:     lnwire.MilliSatoshi(c.testAmt),
-			payAddr: invoice.PaymentAddr,
-		},
-	})
+	c.notifyHtlc(optRHash(hash), optPayAddr(invoice.PaymentAddr),
+		optResolveClose(resolved))
```

Comment:
Looking at these tests in their cleaned-up form, it seems that the distinction between send and close isn't that useful. Instead of close, a normal send and waiting for the send would probably work just as well.

With that change, you could drop both `optResolveClose` and `optResolveSend` and make the `resolved` channel a property of the context too.

Reply:
Done. Big change in readability and compactness. Thanks!

Reply:
Very nice!